File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03620/p03620_num6_parsed.cpp
#include <iostream>
#include <string>

int main() {
    std::string A, B;
    std::cin >> A >> B;
    
    int length = A.length();
    
    int result = -1;
    int operations = length;
    
    for (int i = 0; i < length; i++) {
        int shifts = i;
        int flips = 0;
        
        std::string tempA = A;
        
        for (int j = 0; j < length; j++) {
            if (B[j] == '1') {
                int index = (j + shifts) % length;
             if (tempA[index] == '0') {
                    flips++;
                }
            }
        }
        
        if (tempA == B && flips < operations) {
            result = flips;
            operations = flips;
        }
    }
    
    std::cout << result << std::endl;
    
    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03620/p03620_num6_parsed.py
# Program

def count_operations(A, B):
    if A == B:
        return 0

    n = len(A)
    shifts = []
    flips = []

    for i in range(n):
        if B[i] == '1':
            flips.append(i)

    for shift in range(n):
        A = A[1:] + A[0]
        operations = shift

        for flip in flips:
            A = A[:flip] + str(1 - int(A[flip])) + A[flip+1:]
            operations += 1

        if A == B:
            shifts.append(operations)

    if not shifts:
        return -1

    return min(shifts)


A = input()
B = input()

result = count_operations(A, B)
print(result)